packages/components/src/internal/components/samples/actions.ts

```diff
@@ -370,3 +374,107 @@ export function getDeleteSharedSampleTypeUrl(typeId: number): string {
         returnUrl: window.location.pathname + '#/samples',
     }).toString();
 }
+
+function getSamplesIdsNotFound(queryName: string, orderedIds: string[]) : Promise<string[]> {
+    return new Promise( (resolve, reject) => {
+        Query.selectDistinctRows({
+                schemaName: SCHEMAS.EXP_TABLES.SCHEMA,
+                queryName,
+                column: 'Ordinal',
+                sort: 'Ordinal',
+                success: result => {
+                    const ordinals = result.values
+                    let index = 0;
+                    let oIndex = 0;
+                    let missingIds = [];
+                    // find the gaps in the ordinals values as these correspond to ids we could not find
+                    while (index < orderedIds.length) {
+                        if (oIndex >= ordinals.length || ordinals[oIndex] !== index + 1) {
+                            missingIds.push(orderedIds[index])
+                        } else {
+                            oIndex++;
+                        }
+                        index++;
+                    }
+                    resolve(missingIds);
+                },
+                failure: reason => {
+                    console.error("There was a problem determining the missing Ids", reason);
+                    reject(reason);
+                }
+            });
+    });
+}
+
+export function getFindSamplesByIdData(staleQueryName?: string) : Promise<{ queryName: string, missingIds?: { [key: string]: string[] } }> {
+    return new Promise((resolve, reject) => {
+        const ids = JSON.parse(sessionStorage.getItem(FIND_IDS_SESSION_STORAGE_KEY));
+        if (ids) {
+            Ajax.request({
+                url: ActionURL.buildURL("experiment", "saveOrderedSamplesQuery.api"),
+                method: 'POST',
+                jsonData: {
+                    ids
+                },
+                success: Utils.getCallbackWrapper((response) => {
+                    if (response.success) {
+                        const queryName = response.data;
+                        getSamplesIdsNotFound(queryName, ids).then(notFound => {
+                            let missingIds =  {
+                                [UNIQUE_ID_FIND_FIELD.label]: notFound
+                                    .filter(id => id.startsWith(UNIQUE_ID_FIND_FIELD.storageKeyPrefix))
+                                    .map(id => id.substring(UNIQUE_ID_FIND_FIELD.storageKeyPrefix.length)),
+                                [SAMPLE_ID_FIND_FIELD.label]: notFound
+                                    .filter(id => id.startsWith(SAMPLE_ID_FIND_FIELD.storageKeyPrefix))
+                                    .map(id => id.substring(SAMPLE_ID_FIND_FIELD.storageKeyPrefix.length))
```

Comment:
would it be helpful to have this filter/map sequence in a helper function for jest testing?